server/StudySharp.ApplicationServices/Commands/AddTheoryBlockCommand.cs

```diff
@@ -0,0 +1,49 @@
+using System.Threading;
+using System.Threading.Tasks;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+using StudySharp.Domain.Constants;
+using StudySharp.Domain.General;
+using StudySharp.Domain.Models;
+using StudySharp.DomainServices;
+
+namespace StudySharp.ApplicationServices.Commands
+{
+    public sealed class AddTheoryBlockCommand : IRequest<OperationResult>
+    {
+        public string Name { get; set; }
+        public string Description { get; set; }
+        public int CourseId { get; set; }
+    }
+
+    public sealed class AddTheoryBlockCommandHandler : IRequestHandler<AddTheoryBlockCommand, OperationResult>
+    {
+        private readonly StudySharpDbContext _context;
+
+        public AddTheoryBlockCommandHandler(StudySharpDbContext sharpDbContext)
+        {
+            _context = sharpDbContext;
+        }
+
+        public async Task<OperationResult> Handle(AddTheoryBlockCommand request, CancellationToken cancellationToken)
```

Comment:
Course existence validation is missing

Reply:
Umn
So what is that, in the row 31
_.CourseId == request.CourseId